app/models/seance.rb

```diff
@@ -0,0 +1,28 @@
+class Seance < ApplicationRecord
+  belongs_to :hall
+  belongs_to :movie
+
+  before_validation :end_time
+
+  def end_time
+    self.end_time = self.start_time + self.movie.duration.minutes + 30.minutes
+  end
+
+  validate :is_used
+
+  def is_used
+    Seance.where(hall_id: self.hall_id).each do |item|
+      if self.start_time.between?(item.start_time, item.end_time) ||
+          self.end_time.between?(item.start_time, item.end_time)
+        errors.add(:start_time, "Hall is used for another seance")
+        break
+      end
+    end
```

Comment:
We shouldn't mix active record methods with array methods like you did here, especially when you get some records from the database and then filter elements. You can move checking `start_time` and `end_time` in `where` method :). Then add an error when the array of records is not empty and do nothing when it's empty. `break` is also unneeded then

Reply:
Thank you very much! :)